cors/src/main/java/io/helidon/cors/CorsSupportHelper.java

```diff
@@ -369,14 +370,68 @@ public Aggregator aggregator() {
         return aggregator;
     }
 
-    private boolean isRequestTypeNormal(CorsRequestAdapter<Q> requestAdapter, boolean silent) {
+    // For testing
+    static RequestTypeInfo isRequestTypeNormal(String originHeader, UriInfo requestedHostUri, boolean isRequestSecure) {
+        return RequestTypeInfo.create(originHeader, requestedHostUri, isRequestSecure);
+    }
+
+    /**
+     * Captures intermediate data and the final result in deciding whether a request is a normal (non-CORS) request or not.
+     * <p>
+     *     We want to use the intermediate results for clearer logging if that's turned on without having to recompute it.
+     * </p>
+     * @param originLocation full origin (including port)
+     * @param hostLocation   full host (including scheme and port)
+     * @param isNormal       whether the origin and host information represent a normal (non-CORS) request
+     */
+    record RequestTypeInfo(String originLocation, String hostLocation, boolean isNormal) {
+
+        static RequestTypeInfo create(String originHeader, UriInfo requestedHostUri, boolean isRequestSecure) {
```

Comment:
`isRequestSecure` is ignored? Or my eyes are bad?

Reply:
Good point. It turns out that the `isRequestSecure` method is redundant with how the requested host URI is prepared. That URI always has the correct scheme, so we don't need `isRequestSecure`.

I've removed that method (added in this PR) from the interface and implementations, but I've kept those classes in the PR because I touched them and so I need to make the copyright year 2024 now even though they contain no other changes.